### PR TITLE
DM-40746: Fix size and shape of HiPS Allsky file.

### PR DESCRIPTION
**Pull request by erykoff**
This also adds the ability to specify the Allsky file to be a jpeg or png.

### CODE REVIEW

File: python/lsst/pipe/tasks/hips.py [hunk 1]
```diff
@@ -1549,19 +1575,19 @@ def _write_allsky_file(self, hips_base_path, allsky_order):
             matches = re.match(pixel_regex, png_uri.basename())
             pix_num = int(matches.group(1))
             tile_image = Image.open(io.BytesIO(png_uri.read()))
-            row = math.floor(pix_num//n_tiles_per_side)
-            column = pix_num % n_tiles_per_side
+            row = math.floor(pix_num//n_tiles_wide)
+            column = pix_num % n_tiles_wide
```
Comment (timj) [inline python/lsst/pipe/tasks/hips.py:1579]: Row and column both use the width?

Comment (erykoff) [reply python/lsst/pipe/tasks/hips.py:1579]: Yeah, it's div and mod to get x/y from an ordered list.

**Review by timj — APPROVED** [review]
Looks ok to me.

**Comment by erykoff** [comment]
I have removed the commit which allows you to specify the Allsky filetype since it cannot be specified independently of the rest of the hips images.

**Comment by erykoff** [comment]
Yes indeed, and that's what DM-40747 can do, but it has to be done for everything at once.

**Pull request merged**